jira-mixin/alerts/general.yaml

```diff
@@ -0,0 +1,43 @@
+groups:
+- name: alert.rules
+  rules:
+  - alert: InstanceDown
+    expr: up == 0

```

Comment:
Hmm... `up` is the synthetic metric that'll show whether the exporter is reachable, but probably more useful than that is to know whether the exporter itself can reach JIRA.

I've looking through the exporter code to see if there are any metrics that track that, but I can't find any obvious ones. I _think_ `jira_application_link_status_gauge` might be useful, but I can't figure out if that's relevant...

Reply:
So at first I had tried using the metric `jvm_uptime_gauge`. But when I tested it locally (by making it go into a OOM killed crashloop), it wasn't alerting that it was down. But changing it to `up` then did set off that alert. 

The original metric was
`jvm_uptime_gauge == 0`

Is there a better way to do that that will capture when it's crash looping as well?

Reply:
> Is there a better way to do that that will capture when it's crash looping as well?

Usually alerting on crashlooping pods is done in a general way for all pods using the metrics that come from kube-state-metrics. I was thinking more of the use-case where the exporter is running but is misconfigured and isn't properly reaching its target...

I just realized though that this exporter is a plugin that runs _in_ JIRA as a plugin! That explains why `up` works 😂

In that case it'd probably be best to omit this entirely, and leave up/down detection to something more general. In the past I've configured a generic Prometheus alert something like `sum (up) by (job) / count (up) by (job) < 0.75` - that'll return the ratio of up to discovered and trigger when it's <75%... Either way that part should probably be left to a higher-level more general alert 😉

Reply:
Thanks for the explanation! I removed the InstanceDown alert from the alert yaml and from the README.